### PR TITLE
ShopBreakdown: add `DiscountFieldContainer`

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Add a container for `DiscountField` component to connect it with the API

## Justify why they are needed

- Co-locate related logic with the presentational component

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3010** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3010" target="_blank">Graphite</a> 
    * **PR #3011** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3011" target="_blank">Graphite</a> 
      * **PR #3012** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3012" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3012?utm_source=stack-comment).

**Review by gustaveen — APPROVED** [review]
👌

**Comment by robinandeer** [comment]
@robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3012).

File: apps/store/src/components/ShopBreakdown/DiscountField.tsx [hunk 1]
```diff
@@ -24,8 +24,15 @@ export const DiscountField = (props: Props) => {
   const { t } = useTranslation('cart')
   const [active, setActive] = useState(props.defaultActive ?? false)
 
+  const handleOpenChange = (open: boolean) => {
+    setActive(open)
+    if (!open && props.campaign) {
+      props.onRemove()
+    }
```
Comment (guilhermespopolin) [inline apps/store/src/components/ShopBreakdown/DiscountField.tsx:31]: Nice addition :)

**Review by guilhermespopolin — APPROVED** [review]

**Comment by robinandeer** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3012) rebased this pull request as part of a merge.

**Comment by robinandeer** [comment]
@robinandeer merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3012).

**Pull request merged**